src/engine/cachingreader.cpp

```diff
@@ -200,43 +200,39 @@ CachingReaderChunkForOwner* CachingReader::lookupChunkAndFreshen(SINT chunkIndex
     return pChunk;
 }
 
+// Invoked from the UI thread!!
 void CachingReader::newTrack(TrackPointer pTrack) {
-    // Feed the track to the worker as soon as possible
-    // to get ready while the reader switches its internal
-    // state. There are no race conditions, because the
-    // reader polls the worker.
-    m_worker.newTrack(pTrack);
-    m_worker.workReady();
-    // Don't accept any new read requests until the current
-    // track has been unloaded and the new track has been
-    // loaded.
-    m_state = State::TrackLoading;
-    // Free all chunks with sample data from the current track.
-    freeAllChunks();
+    auto newState = pTrack ? STATE_TRACK_LOADING : STATE_TRACK_UNLOADING;
+    auto oldState = m_state.fetchAndStoreAcquire(newState);
+    VERIFY_OR_DEBUG_ASSERT(
```

Comment:
I guess this can currently actually happen in case the worker is stalled for some reasons. 
We need to prevent it in the engine that this case is really impossible.

Reply:
Wait, from the caching reader perspective this is not longer an issue, since CachingReaderWorker::loadTrack handles the loading and is not interrupted. It is an engine issue, however we cannot judge from m_state, because this is delayed due to the fifos.

Reply:
Ups, the issue is the slotLoadingTrack(TrackPointer pNewTrack, TrackPointer pOldTrack) everywhere. 
New Track is only a wild guess if we allow another new track loading. In most cases pNewTrack is not used but not in every case. We need to clean up this which is risky due to the string based signal slot connections.

So lets stick with this assertion, until we manage to make it happen, but not as critical. 
I will add a comment.